core/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -134,6 +135,28 @@ public static Settings processSettings(Settings settings) {
         return sb.build();
     }
 
+    /**
+     * Interface to allow merging {@link org.elasticsearch.cluster.metadata.MetaData.Custom} in tribe node
+     * When multiple Mergable Custom metadata of the same type is found (from underlying clusters), the
+     * Custom metadata will be merged using {@link #merge(MetaData.Custom)} and the result will be stored
+     * in the tribe cluster state
+     *
+     * @param <T> type of custom meta data
+     */
+    interface MergableCustomMetaData<T extends MetaData.Custom> {
+
+        /**
+         * Merges this custom metadata with other, returning either this or <code>other</code> custom metadata
```

Comment:
nit: can we document that the neither original customs should be changed by this method?